packages/chatbot-server-mongodb-public/src/processors/formatUserMessageForGeneration.ts

```diff
@@ -1,48 +1,57 @@
-/**
-  Adds front matter to the user message text.
- */
-import { ConversationCustomData, logger } from "mongodb-rag-core";
+import {
+  updateFrontMatter,
+  logger,
+  ConversationCustomData,
+} from "mongodb-rag-core";
+import { originCodes } from "mongodb-chatbot-server";
+import { z } from "zod";
+
+const RawCustomDataSchema = z.object({
+  origin: z.string().describe("Origin of the request"),
+  originCode: z
+    .enum(originCodes)
+    // .optional()
+    .default("OTHER")
+    .describe("Code representing the origin of the request"),
+});
 
 export function formatUserMessageForGeneration(
   userMessageText: string,
-  customData?: ConversationCustomData
-) {
-  const frontMatters: Record<string, string> = {};
-  const { origin, originCode } = customData ?? {};
-  if (typeof origin !== "string" || !origin) {
+  customData: ConversationCustomData
+): string {
+  const result = RawCustomDataSchema.safeParse(customData);
+  if (!result.success) {
+    logger.warn(
+      `Invalid customData provided ${customData}`,
+      result.error.message
+    );
```

Comment:
lets tie this to the request. pls use teh logRequest function, passing the reqId